common/app/feed/MostViewed.scala

```diff
@@ -0,0 +1,114 @@
+package feed
+
+import com.gu.commercial.branding.BrandingFinder
+import common.{Edition, GuLogging, editions}
+import contentapi.{ContentApiClient, QueryDefaults}
+import implicits.Requests.RichRequestHeader
+import model.RelatedContentItem
+import services.OphanMostReadItem
+
+import java.net.URL
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+sealed trait Country {
+  val code: String
+  val edition: Edition
+}
+object Country {
+  def fromHeaderString(header: RichRequestHeader): Country =
+    header.countryCode.toLowerCase match {
+      case "gb" => GB
+      case "us" => US
+      case "ca" => CA
+      case "au" => AU
+      case "in" => IN
+      case "ng" => NG
+      case "nz" => NZ
+      case _    => ROW
+    }
+}
+case object GB extends Country {
```

Comment:
@mxdvl [suggested](https://github.com/guardian/frontend/pull/25653#pullrequestreview-1174589222) adding inline comments to the Country objects, to document the names of the countries.

Are these all just ISO alpha-2 country codes (except for `ROW`)?

Also, I feel as though it might be good if we could add a comment explaining why we use this limited list of countries/editions for the MostViewedAgent, but I'm not clear on what the reason actually is. (cc. @jamesgorrie -- I think you had a better grasp of this?)